portfolio-project-matching-app/backend/dao.js

```diff
@@ -848,11 +848,11 @@ const deleteDocAndAssociations = async (coll, id) => {
         for (const map of deleteAssociationsHelper[coll]) {
             const querySnap = await readQuerySnapshotById(map.coll, map.field, id);
             for (const doc of querySnap.docs) {
-                await deleteDoc(map.coll, doc.id);
+                await deleteAssociationById(map.coll, doc.id);
```

Comment:
This should have utilized `deleteAssociation()` all along. I had the two functions swapped on accident.